sjb/actions/multi_sync.py

```diff
@@ -26,15 +29,38 @@ def generate_parameters(self):
         return self.multi.generate_parameters()
 
     def generate_build_steps(self):
-        return self.description() + self.multi.generate_build_steps()
+        return self.description() + self.multi.generate_build_steps() + self.generate_parameter_forwarding_step()
 
     def generate_post_build_steps(self):
         return self.multi.generate_post_build_steps()
 
     def description(self):
         description_lines = ["<div>"]
-        child_descriptions="{}".format("<br/>\n".join([child.description() for child in self.children]))
+        child_descriptions = "{}".format("<br/>\n".join([child.description() for child in self.children]))
         description_lines.append(child_descriptions)
         description_lines.append("</div>")
 
         return [_SYNC_DESCRIPTION_TEMPLATE.render(description="\n".join(description_lines))]
+
+    def generate_parameter_forwarding_step(self):
+        """
+        This is a terrible hack to get around the fact that
+        we take structured data from the configuration and
+        immediately flatten it into XML strings in these
+        generators. A proper approach would keep the data
+        structured and, perhaps, do the conversion to XML
```

Comment:
https://trello.com/c/wXJJd9KL/251-cimaintenance-get-more-out-of-sjb-by-keeping-data-rich